app/src/main/java/layout/mapFragment.java

```diff
@@ -177,23 +179,98 @@ public void onMapReady(GoogleMap map) {
+                view.setLayoutParams(new RelativeLayout.LayoutParams(RelativeLayout.LayoutParams.WRAP_CONTENT, RelativeLayout.LayoutParams.WRAP_CONTENT)); //width, height
+                int index = Integer.parseInt(marker.getSnippet());
+                Mood mood = moodArrayList.get(index);
+                TextView usernameView = (TextView) view.findViewById(R.id.info_window_person);
+                TextView descriptionView = (TextView) view.findViewById(R.id.info_window_description);
+                TextView social_situationView = (TextView) view.findViewById(R.id.info_window_socialSituation);
+                TextView dateView = (TextView) view.findViewById(R.id.info_window_date);
+                TextView feelingView = (TextView) view.findViewById(R.id.info_window_append);
+                ImageView emojiView = (ImageView) view.findViewById(R.id.info_window_emojiImage);
+                ImageView imageView = (ImageView) view.findViewById(R.id.info_window_image);
+
+                usernameView.setText(mood.getUsername());
+                descriptionView.setText(mood.getDescription());
+                social_situationView.setText(mood.getSocialSit());
+                dateView.setText(mood.getDate().toString());
+                feelingView.setText(" - Feeling " + mood.getEmotionalState());
+
+                if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+                    int emojiID = getContext().getApplicationContext().getResources().getIdentifier("emoji" + String.valueOf(mood.getEmoji()),"drawable",getContext().getApplicationContext().getPackageName());
+                    if(emojiID != 0) {
+                        emojiView.setImageResource(emojiID);
+                    }
+                    else { // This field can only be accessed if something goes wrong, or if someone alters the main database. It's mainly a fallback safety.
+                        emojiView.setImageResource(getContext().getApplicationContext().getResources().getIdentifier("err","drawable",getContext().getApplicationContext().getPackageName()));
+                    }
+                }
+                //######################################################
+                String encodedImageString = mood.getImage();
+                if(encodedImageString != null) {
+                    byte[] decodedString = Base64.decode(encodedImageString, Base64.DEFAULT);
+                    Log.d("Bitmap","Length: "+decodedString.length);
+                    Bitmap photo = BitmapFactory.decodeByteArray(decodedString, 0, decodedString.length);
+                    imageView.setImageBitmap(photo);
+                    Log.d("imageTag", "have image");
+                }else {
+                    imageView.setImageBitmap(null);
+                    imageView.setVisibility(View.GONE);
+                    Log.d("imageTag", "no image");
+                }
+
```

Comment:
May need to change how you call imageView.setVisibility(View.GONE) done here. Then again, may not, I'll have to look into this. There's just an issue with resizing windows dynamically like this when adjustViewBounds is "true" on the imageView